### PR TITLE
build: create a duplicate canary-bot deployment to Cloud Run

### LINKED ISSUE #2146: canary-bot: deploy a duplicate instance to Cloud Run
Canary bot should be able to test Cloud Run environments.

See #1817

### PR DESCRIPTION
**Pull request by chingor13**
Towards #1817

Fixes #2146

### CODE REVIEW

File: packages/canary-bot/src/server.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {GCFBootstrapper} from 'gcf-utils';
+import appFn from './canary-bot';
+
+const bootstrap = new GCFBootstrapper();
+const server = bootstrap.server(appFn);
+const port = 8080;
```
Comment (danielbankhead) [inline packages/canary-bot/src/server.ts:20]: Cloud Run provides port for us, which could be different between instances - perhaps we should use `process.env.PORT` while falling back to `8080`:

```ts
const port = process.env.PORT ?? 8080;
```

**Review by danielbankhead — CHANGES_REQUESTED** [review]

File: packages/canary-bot/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+WORKDIR /usr/src/app
+
+COPY package*.json ./
+COPY --from=BUILD /usr/src/app/build build
+RUN npm install --only=production
+
+ENV NODE_ENV "production"
+
+# Run the web service on container startup.
+CMD [ "npm", "run", "start" ]
```
[the remark this answers is not in the document]
Comment (chingor13) [reply packages/canary-bot/Dockerfile]: I assume this is supposed to be `npm --no-update-notifier  run start`?

File: packages/canary-bot/.dockerignore [hunk 3]
```diff
@@ -0,0 +1,4 @@
+build/
+coverage/
+node_modules/
+.env
```
[the remark this answers is not in the document]
Comment (chingor13) [reply packages/canary-bot/.dockerignore:4]: Added

File: packages/canary-bot/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (danielbankhead) [reply packages/canary-bot/Dockerfile]: Either/or: `npm start` is shorthand for `npm run start`

**Review by danielbankhead — APPROVED** [review]

**Comment by chingor13** [comment]
#2201 should allow multiple instances/installations of canary bot to comment on the same issue.

**Review by tmatsuo — APPROVED** [review]

File: packages/canary-bot/cloudbuild.yaml [hunk 4]
```diff
@@ -48,3 +48,34 @@ steps:
...
+      - "-e"
+      - "./scripts/publish-cloud-run.sh"
+      - "$_DIRECTORY"
+      - "$PROJECT_ID"
+      - "$_BUCKET"
+      - "$_KEY_LOCATION"
+      - "$_KEY_RING"
+      - "$_FUNCTION_REGION"
+      # override the bot name
+      - canary-bot-cloud-run
```
Comment (tmatsuo) [inline packages/canary-bot/cloudbuild.yaml:81]: I actually want to deploy the bot with the same name for my loadtest.
Maybe I can create a new bot for loadtesting.

Comment (chingor13) [reply packages/canary-bot/cloudbuild.yaml:81]: For posterity: we discussed adding an option to the publish cloud run script for overriding the queue name.

**Pull request merged**